### ISSUE TITLE
[spark] disallow excluding all fields of a struct

### ISSUE THREAD

**Issue by alancai98** [partiql/partiql-scribe#65]
Spark has somewhat confusing behavior for when empty structs are allowed in different contexts.

```
// In Spark 3.5 and earlier
spark-sql (default)> SELECT STRUCT();
24/08/27 15:56:24 ERROR SparkSQLDriver: Failed in [SELECT STRUCT()]
java.lang.AssertionError: assertion failed: each serializer expression should contain at least one `BoundReference`
```

Similarly for when there's an empty struct within another struct with no other fields:
```
// In Spark 3.5 and earlier
spark-sql (default)> SELECT STRUCT(STRUCT() AS empty_struct_not_top_level) AS top_level;
24/08/27 16:00:04 ERROR SparkSQLDriver: Failed in [SELECT STRUCT(STRUCT() AS empty_struct_not_top_level) AS top_level]
java.lang.AssertionError: assertion failed: each serializer expression should contain at least one `BoundReference`
```

But when the empty struct is in another struct with other fields, there is no error:
```
// In Spark 3.5 and earlier
spark-sql (default)> SELECT STRUCT(STRUCT() AS empty_struct_not_top_level, 'foo' AS other_field) AS top_level;
{"empty_struct_not_top_level":{},"other_field":"foo"}
Time taken: 0.053 seconds, Fetched 1 row(s)
```

Searching the bug reports for Spark, it seems like a known issue https://issues.apache.org/jira/browse/SPARK-46539 that was partially fixed in version 4.0 and later. The first query now works while the second still doesn't work (though has a different error message).
First query now works
```
// In Spark 4.0 and later
spark-sql (default)> SELECT STRUCT();
{}
Time taken: 0.767 seconds, Fetched 1 row(s)
```

Second query still doesn't work
```
// In Spark 4.0 and later
spark-sql (default)> SELECT STRUCT(STRUCT() AS empty_struct_not_top_level) AS top_level;
24/08/27 16:06:05 ERROR SparkSQLDriver: Failed in [SELECT STRUCT(STRUCT() AS empty_struct_not_top_level) AS top_level]
java.lang.AssertionError: assertion failed: each serializer expression should contain at least one `BoundReference` or it should be an empty struct. This is required to ensure that there is a reference point for the serialized object or that the serialized object is intentionally left empty.
```

Third query still works
```
// In Spark 4.0 and later
spark-sql (default)> SELECT STRUCT(STRUCT() AS empty_struct_not_top_level, 'foo' AS other_field) AS top_level;
{"empty_struct_not_top_level":{},"other_field":"foo"}
Time taken: 0.088 seconds, Fetched 1 row(s)
```

So to summarize,
- Spark 3.5 and before does not support empty `STRUCT`s on top-level columns and `STRUCT`s of empty `STRUCT`s
- Spark 3.5 and before does support `STRUCT`s of empty `STRUCT`s that contain other non-empty struct fields
- Spark 4.0 and later supports empty `STRUCT`s on top-level columns
- Spark 4.0 still does not support `STRUCT`s of empty `STRUCT`s (i.e. `SELECT STRUCT(STRUCT() AS empty_struct)`)

IMO, given the somewhat confusing behavior of empty structs in Spark, I think we should disallow it for transpilation much like in Trino. If users really would like to exclude all fields within a struct, just exclude the struct.

E.g. if struct `s` in table `t` had 3 fields, `f1`, `f2`, `f3`, rather than `EXCLUDE t.s.f1, t.s.f2, t.s.f3`, just do `EXCLUDE t.s`.
